homeassistant/components/github/sensor.py

```diff
@@ -150,122 +150,164 @@ def icon(self):
+        self.repository_response = await self._github.repos.get(self.repository_path)
 
-                if open_issues.totalCount > 0:
-                    self.latest_open_issue_url = open_issues[0].html_url
+    async def _update_commit(self):
+        """Update commit data."""
+        self.commit_response = await self._github.repos.list_commits(
+            self.repository_path, **{"params": {"per_page": 1}}
+        )
 
-            latest_commit = repo.get_commits()[0]
-            self.latest_commit_sha = latest_commit.sha
-            self.latest_commit_message = latest_commit.commit.message
+    async def _update_issues(self):
+        """Update issues data."""
+        self.issues_response = await self._github.repos.issues.list(
+            self.repository_path
+        )
 
-            releases = repo.get_releases()
-            if releases and releases.totalCount > 0:
-                self.latest_release_url = releases[0].html_url
+    async def _update_releases(self):
+        """Update releases data."""
+        self.releases_response = await self._github.repos.releases.list(
+            self.repository_path
+        )
 
-            if repo.permissions.push:
-                clones = repo.get_clones_traffic()
-                if clones is not None:
-                    self.clones = clones.get("count")
-                    self.clones_unique = clones.get("uniques")
+    async def _update_clones(self):
+        """Update clones data."""
+        self.clones_response = await self._github.repos.traffic.clones(
+            self.repository_path
+        )
 
-                views = repo.get_views_traffic()
-                if views is not None:
-                    self.views = views.get("count")
-                    self.views_unique = views.get("uniques")
+    async def _update_views(self):
+        """Update views data."""
+        self.views_response = await self._github.repos.traffic.views(
+            self.repository_path
+        )
 
-            self.available = True
-        except self._github.GithubException as err:
-            _LOGGER.error("GitHub error for %s: %s", self.repository_path, err)
-            self.available = False
+    async def _update_pulls(self):
+        """Update pulls data."""
+        response = await self._github.repos.pulls.list(
+            self.repository_path, **{"params": {"per_page": 100}}
+        )
+        if not response.is_last_page:
+            results = await asyncio.gather(
+                *(
+                    self._github.repos.pulls.list(
+                        self.repository_path,
+                        **{"params": {"per_page": 100, "page": page_number}},
+                    )
+                    for page_number in range(
+                        response.next_page_number, response.last_page_number + 1
+                    )
+                )
+            )
+            for result in results:
+                response.data.extend(result.data)
```

Comment:
Is this going to hit a rate limit if there are large number of pulls?

Reply:
yes (and fixing that is planned for another PR)
This is the same behavior as the old (which automatically expanded paginated results), so that issue is not new

Reply:
Well even home assistant doesn't have enough pages for it to be an issue so should be ok

Reply:
yeah, I don't see it really being an issue, but it can and will be better anyway (later)